packages/git/src/browser/diff/git-diff-widget.ts

```diff
@@ -35,6 +39,17 @@ export class GitDiffWidget extends GitBaseWidget<GitFileChangeNode> implements S
 
         this.addClass('theia-git');
 
+        this.toDispose.push(this.gitWatcher.onGitEvent(async gitEvent => {
+            if (GitStatusChangeEvent.is(gitEvent)) {
+                const oldStatus = this.gitStatus;
+                if (!WorkingDirectoryStatus.equals(gitEvent.status, oldStatus)) {
```

Comment:
The backend should [take care](https://github.com/theia-ide/theia/blob/41467de8392ab39fe9641eaa423db191198af4cf/packages/git/src/node/git-repository-watcher.ts#L83) of this, so you do not need to store the working directory state here.

Reply:
I had to do this because otherwise we are ending up in a loop. 
The problem is, that I call the git diff command in setContent. This ends up in https://github.com/theia-ide/theia/blob/41467de8392ab39fe9641eaa423db191198af4cf/packages/git/src/node/git-repository-manager.ts#L37 where we call the repository watcher again initially.
https://github.com/theia-ide/theia/blob/41467de8392ab39fe9641eaa423db191198af4cf/packages/git/src/node/git-repository-watcher.ts#L37
So here, https://github.com/theia-ide/theia/blob/41467de8392ab39fe9641eaa423db191198af4cf/packages/git/src/node/git-repository-watcher.ts#L53, we start a new watch timer, which immediately (because initial is true) fires the statusChanged event. And so it begins again.

Reply:
Passing `initial` as `true` is bogus for syncing.